Engine/QuizEngine/Quiz.swift

```diff
@@ -11,11 +11,12 @@ public final class Quiz {
 		self.flow = flow
 	}
 	
-	public static func start<Delegate: QuizDelegate>(
+    public static func start<Delegate: QuizDelegate, DataSource: QuizDataSource>(
 		questions: [Delegate.Question],
-		delegate: Delegate
+		delegate: Delegate,
+        dataSource: DataSource
 	) -> Quiz where Delegate.Answer: Equatable {
```

Comment:
```suggestion
	) -> Quiz where Delegate.Answer: Equatable, Delegate.Question == DataSource.Question, Delegate.Answer == DataSource.Answer {
```

Reply:
Great! It works.
Thanks for the suggestion!